packages/blocks/src/__internal__/utils/types.ts

```diff
@@ -28,13 +28,24 @@ export interface EditingState {
   rect: DOMRect;
 }
 
-export type DatabaseTableStateType = 'select' | 'clear' | 'click';
-export type DatabaseTableState = {
-  type: DatabaseTableStateType;
+export type DatabaseTableViewRowStateType = 'select' | 'clear' | 'click';
+export type DatabaseTableViewRowState = {
+  type: DatabaseTableViewRowStateType;
   databaseId?: string;
   rowIds?: string[];
 };
 
+export type DatabaseTableViewCellStateType = 'select' | 'clear' | 'edit';
```

Comment:
1. if `Currently only supports single cell selection`, you can make coords just `coords:[CellCoord]` and left TODO there.
2. `DatabaseTableViewCellState` should be divided into 3 types for type narrowing and precise business logic.

examle:
```ts
type CellCoord = {
  rowIndex: number;
  cellIndex: number;
};

type DatabaseTableViewCellSelectState = {
  type: 'select'
  databaseId: string
  coords: [CellCoord]
}
type DatabaseTableViewCellClickState = {
  type: 'click'
  databaseId: string
  coords: [CellCoord]
}
type DatabaseTableViewCellClearState = {
  type: 'clear'
}
```

Reply:
Then, you can remove redundant `?.`